src/UpdatedPackagesCollector.js

```diff
@@ -79,19 +79,40 @@ export default class UpdatedPackagesCollector {
   }
 
   isPackageDependentOf(packageName, dependency) {
-    var dependencies = this.packageGraph.get(packageName).dependencies;
+    if (!this.cache[packageName]) {
+      this.cache[packageName] = {};
+    }
+
+    if (this.cache[packageName][dependency] === "dependent") {
+      return true;
+    } else if (this.cache[packageName][dependency] === "visited") {
+      return false;
+    }
+
+    let dependencies = this.packageGraph.get(packageName).dependencies;
 
     if (dependencies.indexOf(dependency) > -1) {
+      this.cache[packageName][dependency] = "dependent";
       return true;
     }
 
-    return !!find(dependencies, dep => {
```

Comment:
Well it would stop as soon as it found a path where A depended on B. But I guess that doesn't necessarily speed it up because every path is going to be checked anyways.

There has to be a way of representing this data that makes it easy to check every path by simply folding dependencies over. But caching works well enough.